File: mobile/src/core/auth/TokenManager.ts
import * as SecureStore from 'expo-secure-store';

const KEYS = {
  accessToken: 'thingz.accessToken',
  refreshToken: 'thingz.refreshToken',
};

export interface TokenDto {
  accessToken: string;
  refreshToken: string;
}

const TokenManager = {
  async save(dto: TokenDto) {
    await SecureStore.setItemAsync(KEYS.accessToken, dto.accessToken);
    await SecureStore.setItemAsync(KEYS.refreshToken, dto.refreshToken);
  },
  async getAccessToken() {
    return SecureStore.getItemAsync(KEYS.accessToken);
  },
  async getRefreshToken() {
    return SecureStore.getItemAsync(KEYS.refreshToken);
  },
  async isLoggedIn() {
    const token = await SecureStore.getItemAsync(KEYS.accessToken);
    return !!token;
  },
  async clear() {
    await SecureStore.deleteItemAsync(KEYS.accessToken);
    await SecureStore.deleteItemAsync(KEYS.refreshToken);
  },
};

export default TokenManager;

File: ios/ThingzApp/Core/Auth/TokenManager.swift
import Foundation

final class TokenManager {
    static let shared = TokenManager()
    private init() {}

    var accessToken: String? {
        get { KeychainManager.shared.load(for: .accessToken) }
        set {
            if let token = newValue {
                KeychainManager.shared.save(token, for: .accessToken)
            } else {
                KeychainManager.shared.delete(for: .accessToken)
            }
        }
    }

    var refreshToken: String? {
        get { KeychainManager.shared.load(for: .refreshToken) }
        set {
            if let token = newValue {
                KeychainManager.shared.save(token, for: .refreshToken)
            } else {
                KeychainManager.shared.delete(for: .refreshToken)
            }
        }
    }

    var isLoggedIn: Bool { accessToken != nil }

    func save(_ dto: TokenDto) {
        accessToken  = dto.accessToken
        refreshToken = dto.refreshToken
    }

    func clear() {
        KeychainManager.shared.clearAll()
    }
}
